src/OpenTelemetry/Trace/Export/SimpleActivityProcessor.cs

```diff
@@ -41,6 +43,16 @@ public SimpleActivityProcessor(ActivityExporter exporter)
         /// <inheritdoc />
         public override void OnStart(Activity activity)
         {
+            // Perform sampling decision if source is empty.
```

Comment:
this is not 100% correct, as the sampling decision is now made with the current ActivityKind (which would be Internal, the default value). Though none of the built-in Samplers makes use of ActivityKind, nothing prevents one from writing Sampler which takes Kind into consideration.

More better would be to "feed" this activity created outside of `ActivitySource`, into the `ActivityListener`'s `GetRequestedDataUsingContext` pipeline. That can be done in the instrumentationadapter itself, *after* setting the Kind to correct one, (and possibly enrich activity with more tags/links/events etc. if needed)

Reply:
Thanks @CodeBlanch for proposing the idea of "feeding" an external activity to the `GetRequestedDataUsingContext` pipeline.

Reply:
Did you present the idea to .NET team? Any feedback?

Reply:
Given that making sampling decision in SDK piece is going to be incorrect (as we have incorrect Kind), I'll move the sampling logic into the InstrumentationAdapters. Of course this forces Instrumentation to have dependency on Sampling - but doing this to make progress and uncover any other potential issues.

If everything else is good, and the only remaining issue is Instrumentation being aware of Samplers - we'll try to get a new API in activity source to run the `GetRequestedDataUsingContext`.

Reply:
> Given that making sampling decision in SDK piece is going to be incorrect (as we have incorrect Kind)

Is this to set the Kind? reflection can work here till we expose the Kind setter.

Reply:
We can't set the Kind to the correct one in the OT SDK as SDK doesn't know what is the actual kind. InstrumentationAdapter knows it, and it sets it.